synapse/handlers/presence.py

```diff
@@ -209,6 +219,242 @@ async def bump_presence_active_time(self, user: UserID):
         with the app.
         """
 
+    async def update_external_syncs_row(
+        self, process_id, user_id, is_syncing, sync_time_msec
+    ):
+        """Update the syncing users for an external process as a delta.
+
+        This is a no-op when presence is handled by a different worker.
+
+        Args:
+            process_id (str): An identifier for the process the users are
+                syncing against. This allows synapse to process updates
+                as user start and stop syncing against a given process.
+            user_id (str): The user who has started or stopped syncing
+            is_syncing (bool): Whether or not the user is now syncing
+            sync_time_msec(int): Time in ms when the user was last syncing
+        """
+        pass
+
+    async def update_external_syncs_clear(self, process_id):
+        """Marks all users that had been marked as syncing by a given process
+        as offline.
+
+        Used when the process has stopped/disappeared.
+
+        This is a no-op when presence is handled by a different worker.
+        """
+        pass
```

Comment:
I'm a bit confused about why these are being added. They don't seem to be called or defined anywhere in this diff?

Reply:
Err, sorry.  A commit in a different PR needed them to be defined on both to appease mypy, I'll back that out.

Reply:
Oh actually, they are needed now. This is because `hs.get_presence_handler()` returns `BasePresenceHandler` now instead of `PresenceHandler`. So either we a) add a no-op impl here, or b) assert the type in the calling code. I vaguely prefer the former to the latter.

FTR without these stub impls:

```
synapse/replication/tcp/client.py:196: error: "BasePresenceHandler" has no attribute "process_replication_rows"  [attr-defined]
synapse/replication/tcp/handler.py:355: error: "BasePresenceHandler" has no attribute "update_external_syncs_row"  [attr-defined]
synapse/replication/tcp/handler.py:365: error: "BasePresenceHandler" has no attribute "update_external_syncs_clear"  [attr-defined]
```